inc/agent.class.php

```diff
@@ -928,11 +930,23 @@ public function getByTopic($topic) {
             if (strlen($serial)) {
                $computerTable = Computer::getTable();
                $agentTable = self::getTable();
-               return $this->getFromDBByQuery("LEFT JOIN `$computerTable` `c` ON (
-                     `c`.`id` = `$agentTable`.`computers_id`
-                     )
-                     WHERE `$agentTable`.`entities_id`='$entity' AND `c`.`serial` = '$serial'"
-               );
+               $result = $DB->request([
```

Comment:
You forgot to add the WHERE arguments on the new request

Reply:
+1